### PR TITLE
Add sailfish img resize fix to Xperia.

### PR DESCRIPTION
**Pull request by markolemmetty**
Signed-off-by: Marko Lemmetty <marko.lemmetty@jollamobile.com>

### CODE REVIEW

File: kickstart/pack/f5121/hybris [hunk 1]
```diff
@@ -20,43 +20,77 @@ fi
...
 
-# Setup LVM image
-dd if=/dev/zero bs=1 count=0 of=temp.img seek=3000M
+
+IMGSECTORS=0
+IMGBLOCKS=0
+IMGSIZE=0
+BLOCKSIZE=0
+
+function resizeloop {
```
Comment (jusa) [inline kickstart/pack/f5121/hybris]: Perhaps indent the function contents?

File: kickstart/pack/f5121/hybris [hunk 2]
```diff
@@ -20,43 +20,77 @@ fi
...
+/sbin/e2fsck -f -y $LOOP
+
+# Get image blocks and free blocks
+local BLOCKCOUNT=$(/sbin/dumpe2fs -h $LOOP 2>&1 | grep "Block count:" | grep -o -E '[0-9]+')
+local FREEBLOCKS=$(/sbin/dumpe2fs -h $LOOP 2>&1 | grep "Free blocks:" | grep -o -E '[0-9]+')
+BLOCKSIZE=$(/sbin/dumpe2fs -h $LOOP 2>&1 | grep "Block size:" | grep -o -E '[0-9]+')
+echo "$IMG total block count: $BLOCKCOUNT - Free blocks: $FREEBLOCKS"
+
+local IMAGEBLOCKS=$(/usr/bin/expr $BLOCKCOUNT - $FREEBLOCKS)
+local IMAGESECTORS=$(/usr/bin/expr \( $IMAGEBLOCKS \* $BLOCKSIZE \) / 512 )
```
Comment (jusa) [inline kickstart/pack/f5121/hybris]: Unnecessary parentheses.

File: kickstart/pack/f5121/hybris [hunk 3]
```diff
@@ -20,43 +20,77 @@ fi
...
+
+# Shrink to minimum
+echo "Shrink $IMG to $IMAGESECTORS"
+/sbin/resize2fs $LOOP ${IMAGESECTORS}s -f
+
+# Get the size after resize.
+IMGBLOCKS=$(/sbin/dumpe2fs -h $LOOP 2>&1 | grep "Block count:" | grep -o -E '[0-9]+')
+IMGSIZE=$(/usr/bin/expr $IMGBLOCKS \* $BLOCKSIZE)
+IMGSECTORS=$(/usr/bin/expr \( $IMGBLOCKS \* $BLOCKSIZE \) / 512)
```
Comment (jusa) [inline kickstart/pack/f5121/hybris]: Unnecessary parentheses.

File: kickstart/pack/f5121/hybris [hunk 4]
```diff
@@ -20,43 +20,77 @@ fi
...
+ROOTSECTORS=$IMGSECTORS
+
+resizeloop home.img
+
+HOMESIZE=$IMGSIZE
+HOMEBLOCKS=$IMGBLOCKS
+HOMESECTORS=$IMGSECTORS
+
+# We will add some (100M) extra to temp image.
+TEMPIMGSECTORS=$(/usr/bin/expr \( $ROOTSECTORS + $HOMESECTORS + 200000 \) )
```
Comment (jusa) [inline kickstart/pack/f5121/hybris]: Unnecessary parentheses.

File: kickstart/pack/f5121/hybris (hunk 1 above)
Comment (markolemmetty) [reply kickstart/pack/f5121/hybris]: Ah, ok I will change this. Original function resizeloop in ks files (l500d, p4903, 801 etc.) don't have any indent. So I did think that is not used in general. But ok let's fix this. It is more easy to read.

**Pull request merged**